### PR TITLE
Move appKeepers struct to a different package

### PR DESCRIPTION
**Pull request by nguyenvuong1122000**
Closes: #1319 

## What is the purpose of the change

 - Move appKeepers struct to a different package

## Brief change log
 - Create new package called `keepers` at `app/keepers`,
 - Move logic from `app/keepers.go` to new module

## Testing and Verifying

 * This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable

### CODE REVIEW

**Comment by ValarDragon** [comment]
Thanks for making this!

File: app/app.go [hunk 1]
```diff
@@ -192,9 +193,33 @@ func NewOsmosisApp(
 		panic(fmt.Sprintf("error while reading wasm config: %s", err))
 	}
 
-	app.InitSpecialKeepers(skipUpgradeHeights, homePath, invCheckPeriod)
+	app.InitSpecialKeepers(
+		appCodec,
+		bApp,
+		keys,
+		tkeys,
+		memKeys,
```
Comment (ValarDragon) [inline app/app.go:201]: We should be able to generate these in the keepers package right? Perhaps we have a new file in there called storekeys, which has functions for these?

Comment (nguyenvuong1122000) [reply app/app.go:201]: yes, i think generate these keys in keepers package is better.

Comment (nguyenvuong1122000) [reply app/app.go:201]: i think  transfer module should be in keepers package instead of what i did.

File: app/keepers/keepers.go [hunk 2]
```diff
@@ -103,384 +103,401 @@ type appKeepers struct {
...
-	app.ParamsKeeper = &paramsKeeper
+	// transfer module
+	TransferModule transfer.AppModule
 
-	// set the BaseApp's parameter store
-	bApp.SetParamStore(app.ParamsKeeper.Subspace(baseapp.Paramspace).WithKeyTable(paramskeeper.ConsensusParamsKeyTable()))
+	// keys to access the substores
+	keys    map[string]*sdk.KVStoreKey
+	tkeys   map[string]*sdk.TransientStoreKey
+	memKeys map[string]*sdk.MemoryStoreKey
```
Comment (ValarDragon) [inline app/keepers/keepers.go:112]: Oh interesting, right we do still need the map stored. Good call on moving it to this package!

File: app/keepers/keepers.go [hunk 3]
```diff
@@ -508,3 +525,41 @@ func KVStoreKeys() []string {
...
+func (appKeepers *AppKeepers) GetTKey(storeKey string) *sdk.TransientStoreKey {
+	return appKeepers.tkeys[storeKey]
+}
+
+// GetMemKey returns the MemStoreKey for the provided mem key.
+//
+// NOTE: This is solely used for testing purposes.
+func (appKeepers *AppKeepers) GetMemKey(storeKey string) *sdk.MemoryStoreKey {
+	return appKeepers.memKeys[storeKey]
+}
```
Comment (ValarDragon) [inline app/keepers/keepers.go:565]: Perhaps we move this to another file, e.g. `getters.go` or `keys.go` ?

(If going with `keys.go`, we should put the key initialization there as well)

Comment (nguyenvuong1122000) [reply app/keepers/keepers.go]: I moved these functions to `keys.go`. Thanks for reviewing !

File: app/app.go [hunk 4]
```diff
@@ -164,37 +157,40 @@ func NewOsmosisApp(
...
-	keys := sdk.NewKVStoreKeys(KVStoreKeys()...)
-
-	// Define transient store keys
-	tkeys := sdk.NewTransientStoreKeys(paramstypes.TStoreKey)
-
-	// MemKeys are for information that is stored only in RAM.
-	memKeys := sdk.NewMemoryStoreKeys(capabilitytypes.MemStoreKey)
-
 	app := &OsmosisApp{
 		BaseApp:           bApp,
```
Comment (ValarDragon) [inline app/app.go:161]: We should similarly add an AppKeepers empty struct initialization here right?

File: app/keepers/keepers.go [hunk 5]
```diff
@@ -103,384 +103,388 @@ type appKeepers struct {
...
+func (appKeepers *AppKeepers) InitSpecialKeepers(
+	appCodec codec.Codec,
+	bApp *baseapp.BaseApp,
+	wasmDir string,
+	cdc *codec.LegacyAmino,
+	invCheckPeriod uint,
+	skipUpgradeHeights map[int64]bool,
+	homePath string,
+) {
+	appKeepers.GenerateKeys()
```
Comment (ValarDragon) [inline app/keepers/keepers.go:376]: Perhaps we should have an appKeepers.Init() function, and we do this Generate Keys call in there? (Related to comment about initializing the struct in app.go)

**Review by ValarDragon — APPROVED** [review]
LGTM! Nice job on this!

I checked each of the replaces / lines, and it all looked good to me! (Aside from the initailization question for the OsmosisApp struct)

File: app/app.go (hunk 4 above)
Comment (nguyenvuong1122000) [reply app/app.go]: yes, i added it in osmosis app initialization

File: go.mod [hunk 6]
```diff
@@ -224,7 +224,7 @@ require (
 	github.com/stretchr/objx v0.3.0 // indirect
 	github.com/subosito/gotenv v1.2.0 // indirect
 	github.com/sylvia7788/contextcheck v1.0.4 // indirect
-	github.com/syndtr/goleveldb v1.0.1-0.20200815110645-5c35d600f0ca // indirect
+	github.com/syndtr/goleveldb v1.0.1-0.20200815110645-5c35d600f0ca
```
Comment (mattverse) [inline go.mod:227]: What's the reason behind changing goleveldb from inderect -> direct import?

Comment (nguyenvuong1122000) [reply go.mod:227]: I think this change not from my PR. `go.mod` have same change in main branch when i run `go mod tidy`.

**Pull request merged**